### DISCUSSION TITLE
getting the error  "mat1 and mat2 shapes cannot be multiplied (1x800 and 2x5)" in chapter 2 neural network classification.

### DISCUSSION THREAD

**Discussion by RohanDebnath24** [mrdbourke/pytorch-deep-learning#585] [Q&A] [answered]
while making predictions with test samples of x as done in the chapter 02 neural networks classification. I'm getting the error,"mat1 and mat2 shapes cannot be multiplied (1x800 and 2x5)". 
![Screenshot 2023-08-03 215154](https://github.com/mrdbourke/pytorch-deep-learning/assets/141348717/e8b9e2c8-27d3-4c5c-9235-52633e5ebfcf)

**Comment by Surya-Abhinai** [comment]
Can you share the previous code which will help us to find the error

**Comment by mrdbourke** [comment]
Hi @RohanDebnath24 , 

This kind of error is known as a "shape error" and is one of the more common errors in deep learning.

It means there's a shape mismatch between some of your tensors in the previous code.

I'd step back through your values and print out their shapes to see where the potential shape error occurs.

Some helpful resources for shape errors:
- [Shape Errors in PyTorch](https://www.learnpytorch.io/pytorch_most_common_errors/#1-shape-errors-in-pytorch) 
- [Matrix multiplication in PyTorch](https://www.learnpytorch.io/00_pytorch_fundamentals/#matrix-multiplication-is-all-you-need) (this is often the source of different shape errors and is what's happening with your error, two matrices can't be multiplied)

**Comment by alexschedrov** [comment, marked as answer]
The incorrect order of layers in the forward pass of `CircleModelV0` or  `CircleModelV1` causes this error.

Incorrect:
```python
def forward(self, x):
        return self.layer_1(self.layer_2(self.layer_3(x)))
```

Correct:
```python
def forward(self, x):
        return self.layer_3(self.layer_2(self.layer_1(x)))
```

**Reply by mrdbourke** [reply]
Correct, if you get shape errors, it's generally either: 

1. Incorrect inputs/outputs between layers.
2. Incorrect ordering of layers.
3. Incorrect preprocessing inputs into a network.
